client/src/connection/rest/RestLibraryAdapter.ts

```diff
@@ -0,0 +1,248 @@
+// Copyright © 2024, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { AxiosResponse } from "axios";
+
+import { getSession } from "..";
+import {
+  LibraryAdapter,
+  LibraryItem,
+  TableData,
+} from "../../components/LibraryNavigator/types";
+import { LogFn as LogChannelFn } from "../../components/LogChannel";
+import {
+  ColumnCollection,
+  DataAccessApi,
+  RowCollection,
+  TableInfo,
+} from "./api/compute";
+import { getApiConfig } from "./common";
+
+const requestOptions = {
+  headers: { Accept: "application/vnd.sas.collection+json" },
+};
+
+class RestLibraryAdapter implements LibraryAdapter {
+  protected dataAccessApi: ReturnType<typeof DataAccessApi>;
+  protected sessionId: string;
+
+  public constructor(
+    protected readonly emitConnectionNotification: (
+      callback: () => Promise<void>,
+    ) => Promise<void>,
+  ) {}
+
+  public async connect(): Promise<void> {
+    const session = getSession();
+    session.onLogFn = LogChannelFn;
+
+    await this.emitConnectionNotification(async () => await session.setup());
```

Comment:
nit: these 2 lines are still independent of connection type, that is, will be repeated in your itc adapter, right?
If so, how about make it into a base class?
Well actually, I just realized they're also repeated in run command. So perhaps they can be wrapped into the setup function in the base Session class.

Reply:
I'll look at making them part of the base Session class

Reply:
I updated `setup` in the base session class in https://github.com/sassoftware/vscode-sas-extension/pull/733/commits/6b4022f0efc7cb99b92b05911410905188e3b1b8